crates/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -159,6 +168,7 @@ impl DefCollector {
 
         // Must resolve structs before we resolve globals.
         resolve_structs(context, def_collector.collected_types, crate_id, errors);
+        resolve_type_aliases(context, def_collector.collected_type_aliases, crate_id);
```

Comment:
The ordering here means we cannot use type aliases inside of struct definitions. While this is valid Rust:
```rust
type Bar = u32;

struct myStruct {
    foo: Bar,
}

fn main() {
    let s = myStruct { foo: 10u32 };
    println!("{}", s.foo);
}
```

Reply:
This is name resolution whereas collecting names should be an earlier step before this. We should double check that this use case is working as expected though.

Reply:
Yeah, but if you have a Noir program with the same struct as above:
```
type Bar = Field;

struct myStruct {
    foo: Bar,
}
```
you will fail during resolution. When you go to resolve `Bar` which is a named type, the `get_type_alias` method will be called and panic with this: 
```
Message:  no entry found for key
Location: crates/noirc_frontend/src/node_interner.rs:560
```
as the code only pushes to the `type_alias` map during `resolve_type_aliases`: https://github.com/noir-lang/noir/pull/2003/files#r1272800709.

With the way `resolve_type_aliases` is written now the above snippet will pass when it is called before `resolve_structs`.